packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2587,7 +2587,8 @@ export class KeyringController extends BaseController<
     const accounts = await this.#getAccountsFromKeyrings();
 
     switch (type) {
-      case KeyringTypes.simple: {
+      case KeyringTypes.simple:
+      case KeyringTypes.hd: {
```

Comment:
It looks like we are leaving account additions unchecked, which may be also problematic because if a user:
- Imports an SRP
- Imports a private key, part of the above SRP
- Adds an account until reaching the index of the imported private key

User will endup in the same faulty situation